framework/include/base/MooseApp.h

```diff
@@ -616,6 +571,9 @@ class MooseApp :
 
 private:
 
+  /// pointer to the parent/owning MultiApp object if any.
+  MultiApp * _parent_multi = nullptr;
```

Comment:
That is an (legitimate) opinion that I happen to disagree with.  A lot of things become safer, simpler, and easier to understand the way they've been rewritten IMO.  Because MooseApp now takes full ownership of things which it previously counted on outsiders using accessors to do correctly, those things can't be screwed up by other outside code trying to set up some sort of app nesting and forgetting or doing something that breaks assumptions of MooseApp.